clap_complete/tests/snapshots/sub_subcommands.fish

```diff
@@ -8,25 +8,21 @@ complete -c my-app -n "__fish_use_subcommand" -f -a "help" -d 'Print this messag
+complete -c my-app -n "__fish_seen_subcommand_from some_cmd; and not __fish_seen_subcommand_from sub_cmd help" -s h -l help -d 'Print help'
+complete -c my-app -n "__fish_seen_subcommand_from some_cmd; and not __fish_seen_subcommand_from sub_cmd help" -s V -l version -d 'Print version'
+complete -c my-app -n "__fish_seen_subcommand_from some_cmd; and not __fish_seen_subcommand_from sub_cmd help" -f -a "sub_cmd" -d 'sub-subcommand'
+complete -c my-app -n "__fish_seen_subcommand_from some_cmd; and not __fish_seen_subcommand_from sub_cmd help" -f -a "help" -d 'Print this message or the help of the given subcommand(s)'
+complete -c my-app -n "__fish_seen_subcommand_from some_cmd__fish_seen_subcommand_from sub_cmd" -l config -d 'the other case to test' -r -f -a "{Lest quotes\, aren\'t escaped.\t'help,with,comma',Second to trigger display of options\t''}"
+complete -c my-app -n "__fish_seen_subcommand_from some_cmd__fish_seen_subcommand_from sub_cmd" -s h -l help -d 'Print help (see more with \'--help\')'
+complete -c my-app -n "__fish_seen_subcommand_from some_cmd__fish_seen_subcommand_from sub_cmd" -s V -l version -d 'Print version'
+complete -c my-app -n "__fish_seen_subcommand_from some_cmd__fish_seen_subcommand_from help; and not __fish_seen_subcommand_from sub_cmd help" -f -a "sub_cmd" -d 'sub-subcommand'
+complete -c my-app -n "__fish_seen_subcommand_from some_cmd__fish_seen_subcommand_from help; and not __fish_seen_subcommand_from sub_cmd help" -f -a "help" -d 'Print this message or the help of the given subcommand(s)'
+complete -c my-app -n "__fish_seen_subcommand_from some_cmd_alias; and not __fish_seen_subcommand_from sub_cmd help" -s h -l help -d 'Print help'
+complete -c my-app -n "__fish_seen_subcommand_from some_cmd_alias; and not __fish_seen_subcommand_from sub_cmd help" -s V -l version -d 'Print version'
+complete -c my-app -n "__fish_seen_subcommand_from some_cmd_alias; and not __fish_seen_subcommand_from sub_cmd help" -f -a "sub_cmd" -d 'sub-subcommand'
+complete -c my-app -n "__fish_seen_subcommand_from some_cmd_alias; and not __fish_seen_subcommand_from sub_cmd help" -f -a "help" -d 'Print this message or the help of the given subcommand(s)'
+complete -c my-app -n "__fish_seen_subcommand_from some_cmd_alias__fish_seen_subcommand_from sub_cmd" -l config -d 'the other case to test' -r -f -a "{Lest quotes\, aren\'t escaped.\t'help,with,comma',Second to trigger display of options\t''}"
+complete -c my-app -n "__fish_seen_subcommand_from some_cmd_alias__fish_seen_subcommand_from sub_cmd" -s h -l help -d 'Print help (see more with \'--help\')'
+complete -c my-app -n "__fish_seen_subcommand_from some_cmd_alias__fish_seen_subcommand_from sub_cmd" -s V -l version -d 'Print version'
+complete -c my-app -n "__fish_seen_subcommand_from some_cmd_alias__fish_seen_subcommand_from help; and not __fish_seen_subcommand_from sub_cmd help" -f -a "sub_cmd" -d 'sub-subcommand'
```

Comment:
The "help" subcommand could receive a argument named "sub_cmd", but it doesn't really useful at all.
"help" makes the program displays help message and not what `sub_cmd` does.
I think we should strip "help from this list.

Reply:
`foo help` shows `foo --help`

`foo help sub` shows `foo sub --help`

`foo help sub1 sub2` shows `foo sub1 sub2 --help`

Each is meaningful and supporting this was done intentionally, see #4115 and #2315

Reply:
Ah, I see it now. But given the issue raised in https://github.com/clap-rs/clap/pull/4115#issuecomment-1631714950, I believe `foo help help` is not useful enough for a completion.
So to fix the issue could we remove the second "help" from ` not __fish_seen_subcommand_from sub_cmd` conditions.

Reply:
This is getting to too much design discussion for a PR.  As I suggested in that, please open an issue for discussing this (a PR should be reserved for discussing the implementation and we should only start with a PR when things are "obvious").  I'll want more detail on what the problem is and why that is an appropriate fix for it.

Reply:
That's reasonable. I open one in #5555. I'll drop the trouble commit.